modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/IncidentContactCard/components/IncidentContactEditModal/index.jsx

```diff
@@ -51,19 +59,66 @@ const IncidentContactEditModal = ({
 	};
 
 	const handleSubmit = async () => {
-		await Promise.all(
+		try {
+			setIsLoadingSaveButton(true);
+
+			await Promise.all(
+				addHighPriorityContactList?.map((item) => {
+					return addHighPriorityContactsList(client, item);
+				})
+			);
+
+			await Promise.all(
+				addHighPriorityContactList?.map(async (item) => {
+					return associateContactRole(
+						item,
+						project,
+						sessionId,
+						provisioningServerAPI
+					);
+				})
+			);
+
+			await Promise.all(
+				removeHighPriorityContactList?.map((item) => {
+					return removeHighPriorityContactsList(client, item);
+				})
+			);
+			await Promise.all(
+				removeHighPriorityContactList?.map(async (item) => {
+					return removeContactRole(
+						item,
+						project,
+						sessionId,
+						provisioningServerAPI
+					);
+				})
+			);
+
 			removeHighPriorityContactList?.map((item) => {
-				return removeHighPriorityContactsList(client, item);
-			})
-		);
+				openToast(
+					`${item.label}`,
+					`${i18n.translate('high-priority-contact-removed')} 
+						${i18n.translate(`${getKebabCase(item.filter.name)}-contact`)}`
+				);
```

Comment:
I think it is better to capitalize the first character of the user name, do you agree?